ethers-ext/src/core/util.ts

```diff
@@ -43,3 +43,175 @@ export const HexStr = {
+
+  // Account Key Type
+  AccountKeyLegacy = 0x01,
+  AccountKeyPublic = 0x02,
+  AccountKeyFail = 0x03, 
+  AccountKeyWeightedMultiSig = 0x04, 
+  AccountKeyRoleBased = 0x05
+};
+
+
+// For Klay unit
+// https://github.com/ethers-io/ethers.js/blob/main/src.ts/utils/units.ts 
+// https://github.com/klaytn/caver-js/blob/82ba48c52bd3fc2738ccba40f67a4bd0f206822a/packages/caver-utils/src/index.js#L241C1-L420C2
+
+const names = [
+  "peb",
+  "kpeb",
+  "Mpeb",
+  "Gpeb",
+  "ston",
+  "Ston",
+  "uKLAY",
+  "mKLAY",
+  "KLAY",
+  "kKLAY",
+  "MKLAY",
+  "GKLAY",
+  "TKLAY"
+];
+
+const KlayUnit = [
+  { unit: 'peb', pebFactor: 0 },
+  { unit: 'kpeb', pebFactor: 3 },
+  { unit: 'Mpeb', pebFactor: 6 },
+  { unit: 'Gpeb', pebFactor: 9 },
+  { unit: 'ston', pebFactor: 9 },
+  { unit: 'Ston', pebFactor: 9 },
+  { unit: 'uKLAY', pebFactor: 12 },
+  { unit: 'mKLAY', pebFactor: 15 },
+  { unit: 'KLAY', pebFactor: 18 },
+  { unit: 'kKLAY', pebFactor: 21 },
+  { unit: 'MKLAY', pebFactor: 24 },
+  { unit: 'GKLAY', pebFactor: 27 },
+  { unit: 'TKLAY', pebFactor: 30 },
+];
+
+function getFactor(unit: string): number {
+  for ( let i=0; i < KlayUnit.length ; i++ ) {
+    if ( KlayUnit[i].unit === unit) {
+      return KlayUnit[i].pebFactor;
+    }
+  }
+  assertArgument(false, "invalid unit", "unit", unit);
+  return 0;
+}
+
+type Numeric = number | bigint;
+
+/**
+*  Converts %%value%% into a //decimal string//, assuming %%unit%% decimal
+*  places. The %%unit%% may be the number of decimal places or the name of
+*  a unit (e.g. ``"gpeb"`` for 9 decimal places).
+*
+*/
+export function formatKlaytnUnits(value: BigNumberish, unit?: string | Numeric): string {
+  let decimals = 18;
+  if (typeof(unit) === "string") {
+      const index = names.indexOf(unit);
+      assertArgument(index >= 0, "invalid unit", "unit", unit);
+      decimals = getFactor(unit);
+  } else if (unit != null) {
+      decimals = getNumber(unit, "unit");
+  }
+
+  // @ts-ignore
+  return FixedNumber.fromValue( BigNumber.from(value), decimals, { decimals, width: 512 }).toString();
+}
+
+/**
+*  Converts the //decimal string// %%value%% to a BigInt, assuming
+*  %%unit%% decimal places. The %%unit%% may the number of decimal places
+*  or the name of a unit (e.g. ``"gpeb"`` for 9 decimal places).
+*/
+export function parseKlaytnUnits(value: string, unit?: string | Numeric): bigint {
+  assertArgument(typeof(value) === "string", "value must be a string", "value", value);
+
+  let decimals = 18;
+  if (typeof(unit) === "string") {
+      const index = names.indexOf(unit);
+      assertArgument(index >= 0, "invalid unit", "unit", unit);
+      decimals = getFactor( unit );
+  } else if (unit != null) {
+      decimals = getNumber(unit, "unit");
+  }
+
+  // @ts-ignore
+  return FixedNumber.fromString(value, { decimals, width: 512 }).value;
+}
+
+/**
+*  Converts %%value%% into a //decimal string// using 18 decimal places.
+*/
+export function formatKlaytn(peb: BigNumberish): string {
```

Comment:
formatKlay and parseKlay
because we're handling the KLAY currency.

Reply:
Thanks. 
I also found my mistake and already changed.